backend/app/api/routes/users.py

```diff
@@ -205,15 +221,14 @@ def delete_user(
     user = session.get(User, user_id)
     if not user:
         raise HTTPException(status_code=404, detail="User not found")
-    elif user != current_user and not current_user.is_superuser:
+    elif not current_user.is_superuser:
         raise HTTPException(
             status_code=403, detail="The user doesn't have enough privileges"
         )
     elif user == current_user and current_user.is_superuser:
```

Comment:
I can't add the suggestion for the code change above, but the path operation decorator could be:

```Python
@router.delete("/{user_id}", dependencies=[Depends(get_current_active_superuser)])
```

that way it requires the current user to be a superuser and we can remove a couple of `if`s here.

```suggestion
    if user == current_user:
```